lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -315,7 +319,23 @@ async fn process_listener(source_context: SourceContext, listener_context: liste
                 info!("Running a check request: {check_request}");
 
                 match scheduler.run_check(check_request) {
-                    Ok(_) => {}
+                    Ok(_) => {
+                        // drain the global queue and send everything to source_context
+                        let mut event_buffer = source_context.event_buffer_pool().acquire().await;
```

Comment:
I would move this outside of the call to `run_check`, since it's meant to act as a backpressure mechanism to avoid producing/receiving events when we don't have any free event buffers to store them in.